std/uni.d

```diff
@@ -10320,30 +10350,98 @@ private:
     return CodepointSet.fromIntervals(decompressIntervals(compressed));
 }
 
-@safe pure nothrow auto asTrie(T...)(in TrieEntry!T e)
+auto asTrie(A)(in A a)
 {
-    return const(CodepointTrie!T)(e.offsets, e.sizes, e.data);
+    import std.internal.unicode_tables : TrieEntry; // generated file
+    static auto asTrieImpl(T...)(in TrieEntry!T e)
+    {
+        return const(CodepointTrie!T)(e.offsets, e.sizes, e.data);
+    }
+    return asTrieImpl(a);
 }
 
 @safe pure nothrow @nogc @property
 {
-    import std.internal.unicode_tables; // generated file
-
     // It's important to use auto return here, so that the compiler
     // only runs semantic on the return type if the function gets
     // used. Also these are functions rather than templates to not
     // increase the object size of the caller.
```

Comment:
I'm not sure I understand the logic of the comment here. Why would templates increase the object size of the caller if they never get used? And does returning `auto` really suppress the import inside the function body until the function is called?

Reply:
> And does returning auto really suppress the import inside the function body until the function is called?

Yes it does, the compiler needs to have the full function body and run semantic3 on it to determine the return type. Auto return functions behave a lot like templates in that sense, they don't get compiled into the call-site though (AFAIR).